### ISSUE TITLE
References to enum member partially broken since v1.9.5

### ISSUE THREAD

**Issue by sstudi** [doxygen/doxygen#9885]
**Describe the bug**

In the past I referenced to enum members with `\ref EnumName::EnumMember`. This worked fine up to v1.9.4.

Since v1.9.5 it somehow doesn't work anymore. 
I can only reference the enum member by accessing it directly: `\ref EnumMember`.

For `enum class` the behavior is slightly different, but also there `\ref` is no longer working.

**Expected behavior**
Link resolution works by referencing via `\ref EnumName::EnumMember`.

**Screenshots**
If useful, add screenshots to help explain your problem.

**To Reproduce**

Example contains various enums and references.
With 1.9.4 it builds fine and links are working as expected.
[doxygen_enum_reference.zip](https://github.com/doxygen/doxygen/files/10847922/doxygen_enum_reference.zip)

**Version**
v1.9.4 working
v1.9.5 not working
v1.9.6 not working

Windows 10, 64-bit

**Comment by albert-github** [comment]
Looks like we have here a difference between C and C++ code.
I created the following small test program:
```
/// \file
///
/// Some references:
/// 1. classic enum get: \ref classic_enum_get
/// 2. classic enum get: \ref ClassicEnum::classic_enum_get
/// 3. class enum get: \ref class_enum_get
/// 4. class enum get: \ref ClassEnum::class_enum_get
///
/// \include aa.cpp

#include <stdio.h>

/// classic enum
enum ClassicEnum
{
  /// classic get
  classic_enum_get = 0x01
};

/// class enum
enum class ClassEnum
{
  /// class enum get
  class_enum_get = 0x02
};

/// the main function
int main(int argv,char **argc)
{
  printf("1. classic get: %d\n",classic_enum_get);
  printf("2. classic get: %d\n",ClassicEnum::classic_enum_get);
  //printf("3. class enum get: %d\n",class_enum_get);
  printf("4. class enum get: %d\n",ClassEnum::class_enum_get);
  return 0;
}
```
and this gives:
```
aa.cpp:5: warning: unable to resolve reference to 'ClassicEnum::classic_enum_get' for \ref command
aa.cpp:7: warning: unable to resolve reference to 'ClassEnum::class_enum_get' for \ref command
```
but it compiles without problems without problems with gcc (gcc version 11.3.0 (GCC)) and cl (Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29914 for x64).

When creating an analogous c program, i.e. the same source without the `enum  class` and the lines containing references to this `enum class` it still doesn't compile as the `printf` line marked with 2. is not valid.

@doxygen I think best would be to follow the C++ linking and allow links for the lines 1., 2. and 4. what do you think?


Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/10849070/example.tar.gz)

**Comment by doxygen** [comment]
@sstudi @albert-github Please verify if the referenced commit fixes the problem. Do not close the bug report, this will be done automatically when the next official release is out.

**Comment by albert-github** [comment]
I did some quick tests (also with namespace and with class) an it looks OK to me.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
